spec/requests/assign_controller_spec.rb

```diff
@@ -94,11 +94,39 @@
 
   describe "#suggestions" do
     before do
-      SiteSetting.max_assigned_topics = 1
       sign_in(user)
     end
 
+    it "suggests the current user + the last 6 previously assigned users" do
+      assignees = 10.times.map { |_| assign_user_to_post.username }
+
+      get "/assign/suggestions.json"
+
+      suggestions = JSON.parse(response.body)["suggestions"].map { |u| u["username"] }
+      expect(suggestions).to contain_exactly(user.username, *assignees[4..9])
+    end
+
+    it "doesn't suggest users on holiday" do
+      user_on_vacation = assign_user_to_post
+      user_on_vacation.upsert_custom_fields(DiscourseAssign::DiscourseCalendar::HOLIDAY_CUSTOM_FIELD => "t")
+
+      get "/assign/suggestions.json"
+
+      suggestions = JSON.parse(response.body)["suggestions"].map { |u| u["username"] }
```

Comment:
Not sure if you know: `response.parsed_body` instead of doing the `JSON.parse` manually.